ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/CommandService.scala

```diff
@@ -0,0 +1,221 @@
+    "CSduplicateSubmitAndWaitForTransactionId",
+    "SubmitAndWaitForTransactionId returns the same transaction ID for a duplicate submission") {
+    implicit context =>
+      for {
+        alice <- allocateParty()
+        commandId = context.nextCommandId()
+        transactionId1 <- session.bindings.submitAndWaitForTransactionId(
+          alice,
+          context.applicationId,
+          commandId,
+          Seq(Dummy(alice).create.command.command))
+        transactionId2 <- session.bindings.submitAndWaitForTransactionId(
+          alice,
+          context.applicationId,
+          commandId,
+          Seq(Dummy(alice).create.command.command))
+      } yield {
+        assert(
+          transactionId1 == transactionId2,
+          s"The transaction IDs did not match: transactionId1=$transactionId1, transactionId2=$transactionId2")
+      }
+  }
+
+  private val resendingSubmitAndWaitForTransaction = LedgerTest(
+    "CSduplicateSubmitAndWaitForTransaction",
+    "SubmitAndWaitForTransaction returns the same transaction for a duplicate submission") {
+    implicit context =>
+      for {
+        alice <- allocateParty()
+        commandId = context.nextCommandId()
+        transaction1 <- session.bindings.submitAndWaitForTransaction(
+          alice,
+          context.applicationId,
+          commandId,
+          Seq(Dummy(alice).create.command.command))
+        transaction2 <- session.bindings.submitAndWaitForTransaction(
+          alice,
+          context.applicationId,
+          commandId,
+          Seq(Dummy(alice).create.command.command))
+      } yield {
+        assert(
+          transaction1 == transaction2,
+          s"The transaction did not match: transaction1=$transaction1, transaction2=$transaction2")
+      }
+  }
+
+  private val resendingSubmitAndWaitForTransactionTree = LedgerTest(
+    "CSduplicateSubmitAndWaitForTransactionTree",
+    "SubmitAndWaitForTransactionTree returns the same transaction for a duplicate submission") {
+    implicit context =>
+      for {
+        alice <- allocateParty()
+        commandId = context.nextCommandId()
+        transactionTree1 <- session.bindings.submitAndWaitForTransactionTree(
+          alice,
+          context.applicationId,
+          commandId,
+          Seq(Dummy(alice).create.command.command))
+        transactionTree2 <- session.bindings.submitAndWaitForTransactionTree(
+          alice,
+          context.applicationId,
+          commandId,
+          Seq(Dummy(alice).create.command.command))
+      } yield {
+        assert(
+          transactionTree1 == transactionTree2,
+          s"The transaction trees did not match: transactionTree1=$transactionTree1, transactionTree2=$transactionTree2")
+      }
+  }
+
+//  private val submitAndWaitWithInvalidLedgerId = LedgerTest(
+//    "CSsubmitAndWaitInvalidLedgerId",
+//    "SubmitAndWait should fail for invalid ledger ids") { implicit context =>
```

Comment:
We can probably enable this test by allowing to optionally pass a ledger identifier (as suggested before for command identifiers). I would suggest these methods to actually being explicitly named as to make clear that these methods are to be used to generate specific test conditions (i.e. repeating command identifier and invalid ledger identifiers).